### PR TITLE
Update build definitions

### PR DESCRIPTION
**Pull request by weshaggard**
cc @dagood @eerhardt

### CODE REVIEW

File: buildpipeline/pipeline.json [hunk 1]
```diff
@@ -121,6 +124,7 @@
           "Parameters": {
             "PB_TargetArchitecture": "x64",
             "PB_PortableBuild": "true",
+            "PB_AdditionalMSBuildArguments": "/p:SkipTests=$(PB_SkipTests) /p:DotNetRestoreSources=$(PB_RestoreSource)",
```
Comment (eerhardt) [inline buildpipeline/pipeline.json]: Why do we sometimes `-SkipTests` and other times `/p:SkipTests`?

File: buildpipeline/Core-Setup-Windows-Arm-BT.json [hunk 2]
```diff
@@ -76,6 +77,7 @@
     },
     {
       "enabled": true,
+
```
Comment (eerhardt) [inline buildpipeline/Core-Setup-Windows-Arm-BT.json]: unnecessary new line??

**Comment by eerhardt** [comment]
Why aren't you changing any of the Linux build defs?

**Comment by weshaggard** [comment]
>Why aren't you changing any of the Linux build defs? I'd assume we'd need to change for publishing.

The linux definitions already supported configuring the tests so I didn't need to modify them.

File: buildpipeline/pipeline.json (hunk 1 above)
Comment (weshaggard) [reply buildpipeline/pipeline.json]: I did't rework all the definitions so if they were parameters to build.sh/cmd then it is just `-SkipTests` but if they were msbuild args then just `/p:SkipTests`. While I don't like the inconsistency either I didn't want to rework the entire build defintion at this point in time.

**Comment by eerhardt** [comment]
> The linux definitions already supported configuring the tests so I didn't need to modify them.

How about publish?

File: buildpipeline/pipeline.json (hunk 1 above)
Comment (eerhardt) [reply buildpipeline/pipeline.json]: fair enough.  it would be nice to be consistent.

**Comment by weshaggard** [comment]
>How about publish?

For this PR all I did was disable the final publish but I'm trying to decouple the publish logic now to make it more consistent with the other repos and ensure we only do "public" publishing from the publish leg which we can configure. That change is coming next week.

File: buildpipeline/Core-Setup-Windows-Arm-BT.json [hunk 3]
```diff
@@ -62,6 +62,7 @@
       "alwaysRun": false,
       "displayName": "Install Signing Plugin",
       "timeoutInMinutes": 0,
+      "condition": "and(succeeded(), in(variables.PB_SignType, 'real', 'test'))",
```
Comment (dagood) [inline buildpipeline/Core-Setup-Windows-Arm-BT.json]: Later in this file the type is hard-coded to `real`, is this ok or should it use `$(PB_SignType)`?

https://github.com/weshaggard/core-setup/blob/75bed989ccb49da15048108cf1fe8ac379c3e461/buildpipeline/Core-Setup-Windows-Arm-BT.json#L65-L72

Comment (weshaggard) [reply buildpipeline/Core-Setup-Windows-Arm-BT.json]: That is OK and is coming from the microbuild task. We ultimately override it by passing in SignType to our build command so it is never really used.

**Comment by eerhardt** [comment]
> For this PR all I did was disable the final publish but I'm trying to decouple the publish logic now to make it more consistent with the other repos and ensure we only do "public" publishing from the publish leg which we can configure. 

When you do this, make sure all the assets continue going to the exact same places.  We have other pieces that are dependent on the exact location of each asset.

For example, for a single build, the assets go to 2 places.

1. To a folder with the build number `2.1.0-preview1-1234`.
1. To the "channel" folder - i.e for master the folder named "master", or for "release/2.0.0", etc.

We have links and tools that require these assets to be in both places.

**Pull request merged**